### ISSUE TITLE
Fragmented list don’t export animations in PowerPoint

### ISSUE THREAD

**Issue by reparke** [marp-team/marp-cli#141]
(I posted this in Marp VSCode but was told it belonged here instead)

When I used the Fragmented unordered list and export to PPTX, the "fragmented" / animation part doesn't work--that is, all the items appear at once. The fragmented effect works fine in HTML, but not in PPTX
Markdown example:

```
Title
* item 1
* item 2
```

My system
OS: Windows 10 Pro version 1803 (build 17134.885)
VSCode: 1.37.1
Marp extension: 0.9.0

**Comment by yhatt** [comment]
Thank you for your helping to translate the issue from marp-team/marp-vscode#66 to marp-cli.

We have two approaches to realize:

- :a: Add pre-rendered images of all steps provided by fragmented list as new slide pages. It's a realistic appraoch but the page index of exported slides may not match to real pages in Markdown.

- :b: Add step images as slide animations. Most user would expect this behavior. However, [`@marp-team/pptx`](https://github.com/marp-team/pptx) (based on PptxGenJS) is not supported the animation and PptxGenJS is not too. [We also cannot expect the implementation of animation in future version of PptxGenJS.](https://github.com/gitbrent/PptxGenJS/tree/version-3.0#unimplemented-features)

And we have to support rendering images of all animation frames by fragmented list in either case.

**Comment by djmcgreal-cc** [comment]
There's a broader set of issues with the PowerPoint generation than just fragment list animations, for example clickable links. Are there other potential libraries in this space that might generate more dynamic powerpoint content?

**Comment by yhatt** [comment]
@djmcgreal-cc Actually [PptxGenJS](https://gitbrent.github.io/PptxGenJS/) is well-maintained and full-featured library for Node.js. However, we are not using features for dynamic contents (including hyperlink) because reproducing the power of expression about HTML/CSS through them is unreasonably hard. e.g. [HTML to PowerPoint by PptxGenJS](https://gitbrent.github.io/PptxGenJS/html2pptx/) is focusing only to the table.

Marp is focusing to visual reproducibility of generated HTML slide than interactivity of the slide (clickable link, media, animation, etc). Thus, we take an approach to take a picture of slide and set it as the background image. This approach is actually same as [GitPitch](https://medium.com/@gitpitch/gitpitch-embraces-microsoft-powerpoint-keynote-google-slides-7095017dd57c) (shutted down). If you want more interactivity, just using HTML slide is the best.

Another solution to make PPTX with hyperlink and editability from Marp Markdown is already explained in https://github.com/marp-team/marp/discussions/82.

**Comment by djmcgreal-cc** [comment]
Thank you yhatt.

We can present using the web version but must distribute via PowerPoint. As such, links are important - so it's good to know that the PDF to PPT route exists. In this case it needs to be automated in a CI process. Could you recommend a good command line tool that's somewhat the equivalent of https://www.adobe.com/uk/acrobat/online/pdf-to-ppt.html?

**Comment by yhatt** [comment]
No. As already described in marp-team/marp#82, I have never seen FLOSS CLI tool for converting PDF to PPTX.

In addition, there is really few choice even in the commercial product: [Adobe has answered simply "No" for command line solution of PDF -> PPTX via Adobe Acrobat.](https://community.adobe.com/t5/acrobat/how-to-get-xi-pro-to-convert-pdf-to-ppt-at-command-prompt/m-p/8976329#M48892) As far as I saw, any alternative products (Solid Converter, Smallpdf desktop, Foxit PDF editor, Nitro PDF, etc...) also have not the command line interface.

**Comment by junichiyamasaki** [comment]
Hi, I am a beginner of marp, and I have a similar issue for fragmented lists. Animation does not work even in HTML or PDF. The sample code is essentially the same with the above:

```
---
marp: true
---
* test
* test2

```

My system is
- OSX 10.15.7（19H1419）
- Marp for VS Code v1.4.7 
- Visual Studio Code: 1.63.2 (Universal)


Or, do we need any additional setting to enable fragmented lists?

**Comment by yhatt** [comment]
@junichiyamasaki This issue is for discussion about fragmented list in PPTX conversion in Marp CLI. 

If not worked in HTML, please start a new discussion in https://github.com/marp-team/marp/dicsussions, or open a new issue to a corresponding repository (marp-vscode) with filling bug form.

**Comment by junichiyamasaki** [comment]
@yhatt Thank you for the guidance. I have just reported from the marp-vscode bug form.
